src/logging.cpp

```diff
@@ -184,6 +184,90 @@ bool GetLogCategory(BCLog::LogFlags& flag, const std::string& str)
     return false;
 }
 
+std::string LogLevelToStr(BCLog::Level level)
+{
+    switch (level) {
+    case BCLog::Level::None:
+        return "none";
+    case BCLog::Level::Debug:
+        return "debug";
+    case BCLog::Level::Info:
+        return "info";
+    case BCLog::Level::Warning:
+        return "warning";
+    case BCLog::Level::Error:
+        return "error";
+    }
+    assert(false);
+}
+
+std::string LogCategoryToStr(BCLog::LogFlags category)
+{
+    switch (category) {
+    case BCLog::LogFlags::NONE:
```

Comment:
It seems there's an overlap here with `CLogCategoryDesc LogCategories[]`. I think it would be best to use that? If not, at the least it would be good to add comments in both places to keep the lists in sync.

Reply:
> This is due to most of the enum values of LogCategories being bit flags (ie 1 << i) but LogCategories::NONE and LogCategories::ALL are not bit flags.

One way to do it would be to introduce a third "source data" structure, and build both `LogCategories` and the one used for the reverse from that.

That said, I think this is ok for now.